text/0000-kebab-actions.md

```diff
@@ -0,0 +1,202 @@
+or cancel bubbling.
+
+Furthermore, even invoking an action can be inconsistent. For example to call an
+action from a component `sendAction('actionName')` is called. To call an action
+from a controller (and bubble it to the route), `send('actionName')` is used.
+
+By removing the remaining use-case for classic actions, nearly all these
+confusing discrepancies can be eliminated. Ember is easier to learn and master.
+
+# Detailed design
+
+#### Kebab Action Attachment
+
+To add an event listener to a DOM node, the author may use "kebab" syntax. By
+this we mean the dasherized form of an event name, prefixed by `on-`.
+
+For example to attach a `click` event to a button, the following syntax would
+be used:
+
+```hbs
+<button on-click={{action 'save'}}>Save</button>
+```
+
+For comparison, to attach the `save` action with classic actions this syntax
+would be used instead:
+
+```hbs
+<button {{action 'save'}}>Save</button>
+```
+
+Kebabs are more verbose in the above case, but have the
+advantage of a clear and extensible syntax. Consider attaching a
+`mouseover` event:
+
+```hbs
+<div on-mouse-over={{action 'save'}}>Save</div>
+```
+
+vs.
+
+```hbs
+<div {{action 'save' on='mouseOver'}}>Save</div>
+```
+
+The kebab version is an basic evolution of the `on-click` usage.
+
+The second advantage of kebab syntax is its parity with Ember component
+action passing. For example consider these components:
+
+```hbs
+{{my-button on-click=(action 'save')}}Save{{/my-button}}
+<my-button on-click={{action 'save'}}>Save</my-button>
+```
+
+Kebab syntax is much closer to the above than classic action syntax. Of course
+a kebab action passed to a component (or angle bracket component) is simply
+a function with no special behavior.
+
+#### Handling a Kebab Action
+
+Kebab actions will be invoked with the raw DOM event (not the jQuery event)
+as an argument. For example:
+
+```js
+// app/components/my-button.js
+export Ember.Component.extend({
+  actions: {
+    save(event) {
+      console.log('saved with click on ', event.target);
+      this.get('model').save();
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-button.hbs }}
+<button on-click={{action 'save'}}>Save</button>
+```
+
+Actions curry arguments, thus the event may not always be the first argument. For
+example:
+
+```js
+// app/components/my-input.js
+export Ember.Component.extend({
+  actions: {
+    log(prefix, event) {
+      console.log(prefix, Ember.$(event.target).val());
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-input.hbs }}
+<input on-change={{action 'log' 'value is:'}} />
+```
+
+Additionally, the `value` argument to `action` is useful when passing the
+event.
+
+
+```js
+// app/components/my-input.js
+export Ember.Component.extend({
+  actions: {
+    log(value) {
+      console.log(target);
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-input.hbs }}
+<input on-change={{action 'log' value='target.value'}} />
```

Comment:
Yeah I think that is uncontroversial, just needs to happen. I suppose it should happen with these changes just for ergonomics, since this is where `valuePath=` should become more commonly used.